docs/products/opensearch/howto/opensearch-search-and-python.rst

```diff
@@ -0,0 +1,471 @@
+
+Search lite API and query DSL
+-----------------------------
+There are two ways of performing search queries in OpenSearch: `Search Lite API <https://opensearch.org/docs/1.2/opensearch/rest-api/search/>`_ and `OpenSearch query domain-specific language (DSL) <https://opensearch.org/docs/latest/opensearch/query-dsl/index/>`_.
+In the Search Lite API, it is used the ``q`` parameter to run a query parameter search. But those kinds of queries do not support the Query DSL, and they work better for simple searches because it is expected a string as a parameter, for example:
+
+.. code-block:: python
+
+    client.search({
+        index: 'recipes',
+        q: 'ingredients:broccoli AND calories:(>=100 AND <200)'
+    })
+
+
+With Query DSL, the field ``body`` expects a dictionary object which makes it easier to construct more complex search queries. As an example:
+
+.. code-block:: python
+
+     query_body = {
+                    "query": {
+                      "multi_match": {
+                        "query": "Garlic-Lemon",
+                        "fields": [
+                          "title",
+                          "ingredients"
+                        ]
+                      }
+                    }
+                  }
+    resp = client.search(index=INDEX_NAME, body=query_body)
+    log_titles(resp)
+
+In this example, we are searching for "Garlic-Lemon" across ``title`` and ``ingredients`` fields.
+
+.. note::
+  In the method ``search()``. One of the optional fields is the ``size`` field, which is defined as the number of results returned in the search. The default value is 10.
+  
+
+We are not adjusting the ``size`` parameter in this tutorial. Therefore, we are using the default value which is 10 results per search.
+
+
+Common queries
+''''''''''''''
+
+In the next section, we will cover some of the more common queries. Let's start querying 🔎 
+
+Match query
+-----------
+
+The ``match`` query helps you to find the best matches with multiple search word and is the default option for full-text search. For example, if you want to find matches that in the ``title`` has "Chilled Tomato".
+This will return results of titles that contain "Chilled" or "Tomato" on it due to DSL defaults to the "or" operator.
+
+.. code-block:: python
+       query_body = {
+                      "query": {
+                        "match": {
+                          "title": "Chilled Tomato"
+                        }
+                      }
+                    }
+      resp = client.search(index=INDEX_NAME, body=query_body)
+      log_titles(resp)
+
+If you want to find exact matches for fields in the ``title`` as "Chilled Tomato", you can specify the operator "and".
+
+.. code-block:: python
+
+       query_body = {
+                      "query": {
```

Comment:
Looking at the examples I'm still thinking that for a reader seeing some generic way of writing an example (maybe even without specific search terms) + a command to execute the example from your demo project would be the easiest way to read and then to replicate for their projects.

However, I understand that I suggest same way as I did when writing the article for NodeJS and I'm biased :D

Reply:
It is okay. I think is a valuable insight and I also think is a good idea. I've cleaned up